app/javascript/controllers/image_upload_controller.js

```diff
@@ -1,56 +1,80 @@
-      this.preventDefaults(event),
-    )
-    this.fileTarget.addEventListener("dragenter", (event) =>
-      this.preventDefaults(event),
-    )
-    this.fileTarget.addEventListener("drop", (event) => this.drop(event))
+  disconnect() {
+    this.fileTarget.removeEventListener("change", this.boundPreviewUpdate)
+    this.contentTarget.removeEventListener("dragover", this.boundIgnore)
+    this.contentTarget.removeEventListener("dragenter", this.boundIgnore)
+    this.contentTarget.removeEventListener("drop", this.boundDrop)
+    this.contentTarget.removeEventListener("paste", this.boundPasteHandler)
+  }
+
+  boundPreviewUpdate = () => { this.previewUpdate() }
+  previewUpdate() {
+    const input = this.fileTarget
+    if (input.files && input.files[0]) {
+      const reader = new FileReader()
+      reader.onload = (e) => {
+        this.previewTarget.querySelector("img").src = e.target.result
+        this.element.classList.add("show-previews")
+        this.contentTarget.focus()
+        window.dispatchEvent(new Event('resize')) // Throw this event will cause textarea_autogrow to reprocess
+      }
+      reader.readAsDataURL(input.files[0])
+    }
   }
 
-  preventDefaults(event) {
+  previewRemove() {
+    this.previewTarget.querySelector("img").src = ''
+    this.element.classList.remove("show-previews")
+    this.contentTarget.focus()
+    window.dispatchEvent(new Event('resize')) // Throw this event will cause textarea_autogrow to reprocess
+  }
+
+  boundIgnore = (event) => { this.ignore(event) }
+  ignore(event) {
     event.preventDefault()
     event.stopPropagation()
   }
 
+  boundDrop = (event) => { this.drop(event) }
   drop(event) {
     this.preventDefaults(event)
     let files = event.dataTransfer.files
     this.fileTarget.files = files
   }
 
-  pasteHandler() {
-    this.contentTarget.addEventListener("paste", async (event) => {
-      const clipboardData =
-        event.clipboardData || event.originalEvent.clipboardData
+  boundPasteHandler = async (event) => { this.pasteHandler(event) }
+  async pasteHandler(event) {
+    const clipboardData =
+      event.clipboardData || event.originalEvent.clipboardData
 
-      for (const item of clipboardData.items) {
-        if (item.kind === "file") {
-          const blob = item.getAsFile()
-          if (!blob) return // return if no valid blob
+    for (const item of clipboardData.items) {
+      if (item.kind === "file") {
+        const blob = item.getAsFile()
+        if (!blob) return
 
-          try {
-            const dataURL = await this.readBlobAsDataURL(blob)
-            this.addImageToFileInput(dataURL, blob.type)
-          } catch (error) {
-            console.error("Error reading pasted image:", error)
-          }
+        try {
+          const dataURL = await this.readPastedBlobAsDataURL(blob)
+          this.addImageToFileInput(dataURL, blob.type)
+        } catch (error) {
+          console.error("Error reading pasted image:", error)
```

Comment:
Why not just let this error throw out to the console on it's own? Does catching here allow you to continue doing other things?

Reply:
Good call, this was old code from Ali that I didn't correct. I'll update this.